src/plugins/discover/public/application/view_alert/view_alert_route.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useEffect, useMemo } from 'react';
+import { useHistory, useLocation, useParams } from 'react-router-dom';
+import { sha256 } from 'js-sha256';
+import { i18n } from '@kbn/i18n';
+import { ToastsStart } from 'kibana/public';
+import type { Alert } from '../../../../../../x-pack/plugins/alerting/common';
+import { AlertTypeParams } from '../../../../../../x-pack/plugins/alerting/common';
+import { getTime, SerializedSearchSourceFields } from '../../../../data/common';
+import type { Filter, TimeRange } from '../../../../data/public';
+import { MarkdownSimple, toMountPoint } from '../../../../kibana_react/public';
+import { DiscoverAppLocatorParams } from '../../locator';
+import { useDiscoverServices } from '../../utils/use_discover_services';
+
+interface SearchThresholdAlertParams extends AlertTypeParams {
+  searchConfiguration: SerializedSearchSourceFields;
+}
+
+interface QueryParams {
+  from: string | null;
+  to: string | null;
+  checksum: string | null;
+}
+
+type NonNullableEntry<T> = { [K in keyof T]: NonNullable<T[keyof T]> };
+
+const LEGACY_BASE_ALERT_API_PATH = '/api/alerts';
+const DISCOVER_MAIN_ROUTE = '/';
+
+const displayError = (title: string, errorMessage: string, toastNotifications: ToastsStart) => {
+  toastNotifications.addDanger({
+    title,
+    text: toMountPoint(<MarkdownSimple>{errorMessage}</MarkdownSimple>),
+  });
+};
+
+const displayRuleChangedWarn = (toastNotifications: ToastsStart) => {
+  const warnTitle = i18n.translate('discover.viewAlert.alertRuleChangedWarnTitle', {
+    defaultMessage: 'Alert rule has changed',
+  });
+  const warnDescription = i18n.translate('discover.viewAlert.alertRuleChangedWarnDescription', {
+    defaultMessage: `Displayed documents might not match the documents triggered notification, 
+    since the rule configuration has been changed.`,
+  });
+
+  toastNotifications.addWarning({
+    title: warnTitle,
+    text: toMountPoint(<MarkdownSimple>{warnDescription}</MarkdownSimple>),
+  });
+};
```

Comment:
I think what's missing here is the notification when alert rule hasn't changed, since docs could also differ:

As written in #117532

`If the checksum matches a different notification should be displayed, because the displayed documents might also be different to the documents that triggered the alert due to more ingested, indexed or deleted documents`

In this case it should not be a warning, but a info notification, like this

Title: `Possibly different documents`
Description: `Displayed documents might not match the documents triggered notification, 
     since documents might have been deleted or added.`

Does this sound ok? Dear @gchaps

Reply:
Applied this change, see description.

Reply:
Thx @gchaps , FYI @Dmitriynj, pls resolve conversation when it's done, PR is getting loooooong